modules/api/php/views/project/instruments.class.inc

```diff
@@ -69,7 +69,9 @@ class Instruments
                 'Subgroup'               => $instrument->getSubgroupname(),
                 'DoubleDataEntryEnabled' => $instrument->isDDE(),
             );
-            $instruments[$shortname] = $item;
+            if (!is_null($shortname)) {
```

Comment:
I don't see any reason this should be possible. This is going to cause silent failures down the line.

Reply:
https://github.com/aces/Loris/blob/4155ea37030eecc913701b1aebd81e0798916dbb/modules/api/php/models/projectinstrumentsrow.class.inc#L50-L58

Reply:
The shortname is used as a key in the return response. This silently removes instruments from the response. I don't know why the projectinstrumentsrow does that, an instrument should always have a shortname.

Reply:
Should I modify projectsinstrumentsrow then?

Reply:
@xlecours do you know why getShortname in projectinstrumentsrow is nullable?

Reply:
@driusan my guess is that the Name column is nullable in the project table:
```
`Name` VARCHAR(255) NULL,
```